### PR TITLE
GitHub webhook service

### PR DESCRIPTION
**Pull request by rimolive**

### CODE REVIEW

File: services/github-service/src/main/java/org/kontinuity/catapult/service/github/api/NoSuchWebhookEventException.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package org.kontinuity.catapult.service.github.api;
+
+/**
+ * Indicates a specified Webhook event does not exist.
+ * 
+ * @author <a href="mailto:rmartine@redhat.com">Ricardo Martinelli de Oliveira</a>
+ */
+public class NoSuchWebhookEventException extends RuntimeException {
```
Comment (ALRubinger) [inline services/github-service/src/main/java/org/kontinuity/catapult/service/github/api/NoSuchWebhookEventException.java]: Is this a result of input from the user?  If so maybe extends IllegalArgumentException?

File: services/github-service/src/main/java/org/kontinuity/catapult/service/github/impl/kohsuke/KohsukeGitHubWebhook.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	}
+
+	@Override
+	public String getUrl() {
+		return delegate.getUrl().toString();
+	}
+
+	@Override
+	public GitHubWebhookEvent[] getEvents() {
+		return null; //TODO fix this return
```
Comment (ALRubinger) [inline services/github-service/src/main/java/org/kontinuity/catapult/service/github/impl/kohsuke/KohsukeGitHubWebhook.java]: Still TODO?

File: services/github-service/src/test/java/org/kontinuity/catapult/service/github/impl/kohsuke/GitHubServiceIT.java [hunk 3]
```diff
@@ -75,11 +75,16 @@ public void cannotForkNonexistentRepo(){
...
+    	
     	final GitHubRepository targetRepo = gitHubService.fork(NAME_GITHUB_SOURCE_REPO);
-    	gitHubService.createWebHook(
-    			targetRepo.getFullName(),
-    			"https://10.1.2.2",						// TODO Webhook URL
-    			new KohsukeGithubWebhookEvent("ALL"));
+		GitHubWebhook webhook = gitHubService.createWebhook(
+    			targetRepo,
+    			webhookUrl,
+    			new KohsukeGitHubWebhookEvent("ALL"));
```
Comment (ALRubinger) [inline services/github-service/src/test/java/org/kontinuity/catapult/service/github/impl/kohsuke/GitHubServiceIT.java]: Can't have the user calling impl code.  Probably we want some API way of getting all WebhookEvents as an enum, yes?  Limits user error by freefloating Strings.

Comment (rimolive) [inline services/github-service/src/test/java/org/kontinuity/catapult/service/github/impl/kohsuke/GitHubServiceIT.java]: Correct, so I'm converting GitHubWebhookEvent to an enum and removing KohsukeGitHubWebhookEventImpl.

File: services/github-service/src/main/java/org/kontinuity/catapult/service/github/impl/kohsuke/KohsukeGitHubWebhook.java (hunk 2 above)
Comment (rimolive) [inline services/github-service/src/main/java/org/kontinuity/catapult/service/github/impl/kohsuke/KohsukeGitHubWebhook.java]: Doh! /o\

File: services/github-service/src/main/java/org/kontinuity/catapult/service/github/api/NoSuchWebhookEventException.java (hunk 1 above)
Comment (rimolive) [inline services/github-service/src/main/java/org/kontinuity/catapult/service/github/api/NoSuchWebhookEventException.java]: It should be used to return to the user that he sent an invalid event, but with the change of GitHubWebhookEvent to an enum this Exception is useless now. I'm removing it.

**Comment by ALRubinger** [comment]
@rimolive I have squashed your commits, added some work of my own, rebased atop master, and pushed that result into upstream/40.  https://github.com/redhat-kontinuity/catapult/commits/40  Can you use that as your new branch point and let's address the following 2 issues before this goes upstream/master?

1) Test failure:

```
Running org.kontinuity.catapult.service.github.impl.kohsuke.GitHubServiceIT
Tests run: 5, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 1.73 sec <<< FAILURE! - in org.kontinuity.catapult.service.github.impl.kohsuke.GitHubServiceIT
createGithubWebHook(org.kontinuity.catapult.service.github.impl.kohsuke.GitHubServiceIT)  Time elapsed: 0.249 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<https://[10.1.2.2]> but was:<https://[api.github.com/repos/RedHatDistortion/jboss-eap-quickstarts/hooks/8074994]>
	at org.kontinuity.catapult.service.github.impl.kohsuke.GitHubServiceIT.createGithubWebHook(GitHubServiceIT.java:79)
````

2) When running the test twice, it will fail on creation of the webhook because the webhook already exists.  Recommend we, in the test:
* Create a new repository with unique name using the GitHubService
* Create a webhook
* On test cleanup, delete the repository we just created

There are examples of the above in OpenShiftServiceIT where we introduce an SPI for us, in tests, to delete things, but we don't expose this to the user API.

With new commits to work atop upstream/40 that satisfy the above, we can merge to master.

**Comment by ALRubinger** [comment]
Upstream in 64f2176a544d887d5eaf73725145133bca2bc31f

**Pull request closed without merging**